fastly/validators.go

```diff
@@ -34,29 +34,108 @@ func validateLoggingMessageType(v interface{}, k string) (ws []string, errors []
 
 func validateLoggingPlacement(v interface{}, k string) (ws []string, errors []error) {
 	value := v.(string)
-	validTypes := map[string]struct{}{
+	validPlacements := map[string]struct{}{
 		"none":      {},
 		"waf_debug": {},
 	}
 
-	if _, ok := validTypes[value]; !ok {
+	if _, ok := validPlacements[value]; !ok {
 		errors = append(errors, fmt.Errorf(
 			"%q must be one of ['none', 'waf_debug']", k))
 	}
 	return
 }
 
+func validateDirectorQuorum(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(int)
+	if value < 0 || value > 100 {
+		errors = append(errors, fmt.Errorf(
+			"%q must be a percentage between 0 and 100", k))
+	}
+	return
+}
+
 func validateDirectorType(v interface{}, k string) (ws []string, errors []error) {
 	value := uint(v.(int))
-	validVersions := map[uint]struct{}{
+	validTypes := map[uint]struct{}{
 		1: {},
 		3: {},
 		4: {},
 	}
 
-	if _, ok := validVersions[value]; !ok {
+	if _, ok := validTypes[value]; !ok {
 		errors = append(errors, fmt.Errorf(
 			"%q must be one of ['1' (random), '3' (hash), '4' (client)]", k))
 	}
 	return
 }
+
+func validateConditionType(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	validTypes := map[string]struct{}{
+		"REQUEST":  {},
+		"RESPONSE": {},
+		"CACHE":    {},
+	}
+
+	if _, ok := validTypes[value]; !ok {
+		errors = append(errors, fmt.Errorf(
+			"%q must be one of ['REQUEST', 'RESPONSE', 'CACHE']", k))
```

Comment:
Could this message be built using the values in the previous data structure, (or string array, see last comment)?  Just a way to keep them in sync.